extensions/notebook/src/dialog/addNotebookDialog.ts

```diff
@@ -0,0 +1,179 @@
+import * as azdata from 'azdata';
+import * as path from 'path';
+import { pathExists, remove } from 'fs-extra';
+import * as loc from '../common/localizedConstants';
+import { IconPathHelper } from '../common/iconHelper';
+import { FileType, IBookTocManager } from '../book/bookTocManager';
+import { confirmReplace, getDropdownValue } from '../common/utils';
+import { IPrompter } from '../prompts/question';
+import CodeAdapter from '../prompts/adapter';
+import { BookModel } from '../book/bookModel';
+import { BookTreeItemType } from '../book/bookTreeItem';
+
+export class AddNotebookDialog {
+	private dialog: azdata.window.Dialog;
+	public view: azdata.ModelView;
+	private formModel: azdata.FormContainer;
+	private notebookNameInputBox: azdata.InputBoxComponent;
+	private booksDropdown: azdata.DropDownComponent;
+	private notebookTypeDropdown: azdata.DropDownComponent;
+	private saveLocationInputBox: azdata.InputBoxComponent;
+	private prompter: IPrompter;
+	private bookNames: string[];
+
+	constructor(private tocManager: IBookTocManager, private books: BookModel[]) {
+		this.prompter = new CodeAdapter();
+		this.bookNames = this.books.length > 0 ? this.books.map(b => b.bookItems[0].title) : [];
+		this.bookNames.unshift('-');
+	}
+
+	protected createHorizontalContainer(view: azdata.ModelView, items: azdata.Component[]): azdata.FlexContainer {
+		return view.modelBuilder.flexContainer().withItems(items, { CSSStyles: { 'margin-right': '5px', 'margin-bottom': '10px' } }).withLayout({ flexFlow: 'row', alignItems: 'center' }).component();
+	}
+
+	public async selectFolder(): Promise<string | undefined> {
+		const allFilesFilter = loc.allFiles;
+		let filter: any = {};
+		filter[allFilesFilter] = '*';
+		let uris = await vscode.window.showOpenDialog({
+			filters: filter,
+			canSelectFiles: false,
+			canSelectMany: false,
+			canSelectFolders: true,
+			openLabel: loc.labelSelectFolder
+		});
+		if (uris && uris.length > 0) {
+			return uris[0].fsPath;
+		}
+		return undefined;
+	}
+
+	public async validatePath(folderPath: string, fileBasename: string): Promise<boolean> {
+		const destinationUri = path.join(folderPath, fileBasename);
+		if (await pathExists(destinationUri)) {
+			const doReplace = await confirmReplace(this.prompter, loc.confirmReplaceFile);
+			if (doReplace) {
+				//remove folder if exists
+				await remove(destinationUri);
+				return true;
+			}
+			return false;
+		}
+		return await pathExists(folderPath);
+	}
+
+	public async createDialog(): Promise<void> {
+		this.dialog = azdata.window.createModelViewDialog(loc.newNotebook);
+		this.dialog.registerContent(async view => {
+			this.view = view;
+
+			const notebookLabel = this.view.modelBuilder.text()
+				.withProperties({
+					value: loc.notebookDescription,
+					CSSStyles: { 'margin-bottom': '0px', 'margin-top': '0px', 'font-size': 'small' }
+				}).component();
+
+			this.notebookNameInputBox = this.view.modelBuilder.inputBox()
+				.withProperties({
+					values: [],
```

Comment:
You should be able to just remove the values and value entries if you're not setting them to any initial value